exchanges/gateio/gateio_websocket.go

```diff
@@ -723,14 +712,14 @@ func (g *Gateio) handleSubscription(event string, channelsToSubscribe subscripti
 	return errs
 }
 
-func (g *Gateio) generatePayload(event string, channelsToSubscribe subscription.List) ([]WsInput, error) {
+func (g *Gateio) generatePayload(ctx context.Context, conn stream.Connection, event string, channelsToSubscribe subscription.List) ([]WsInput, error) {
 	if len(channelsToSubscribe) == 0 {
 		return nil, errors.New("cannot generate payload, no channels supplied")
 	}
 	var creds *account.Credentials
 	var err error
 	if g.Websocket.CanUseAuthenticatedEndpoints() {
-		creds, err = g.GetCredentials(context.TODO())
+		creds, err = g.GetCredentials(ctx)
```

Comment:
Yeah as long as the connection accepts auth it will subscribe to it, the only issue is outbound messaging which I will work on when I integrate an exchange which supports it and this design is stable if that is okay.